packages/gestalt/src/Touchable.js

```diff
@@ -73,126 +82,167 @@ const getRoundingStyle = (r: Rounding): Style => {
   return identity();
 };
 
-export default class Touchable extends React.Component<Props> {
-  static propTypes = {
-    accessibilityControls: PropTypes.string,
-    accessibilityExpanded: PropTypes.bool,
-    accessibilityHaspopup: PropTypes.bool,
-    accessibilityLabel: PropTypes.string,
-    children: PropTypes.node,
-    disabled: PropTypes.bool,
-    fullHeight: PropTypes.bool,
-    fullWidth: PropTypes.bool,
-    mouseCursor: PropTypes.oneOf([
-      'copy',
-      'grab',
-      'grabbing',
-      'move',
-      'noDrop',
-      'pointer',
-      'zoomIn',
-      'zoomOut',
-    ]),
-    onBlur: PropTypes.func,
-    onFocus: PropTypes.func,
-    onTouch: PropTypes.func,
-    onMouseEnter: PropTypes.func,
-    onMouseLeave: PropTypes.func,
-    rounding: RoundingPropType,
-  };
-
-  handleKeyPress = (event: SyntheticKeyboardEvent<HTMLDivElement>) => {
-    const { disabled, onTouch } = this.props;
-    // Check to see if space or enter were pressed
-    if (
-      !disabled &&
-      onTouch &&
-      (event.charCode === SPACE_CHAR_CODE || event.charCode === ENTER_CHAR_CODE)
-    ) {
-      // Prevent the default action to stop scrolling when space is pressed
-      event.preventDefault();
-      onTouch({ event });
-    }
-  };
-
-  handleBlur = (event: SyntheticFocusEvent<HTMLDivElement>) => {
-    const { disabled, onBlur } = this.props;
-    if (!disabled && onBlur) {
-      onBlur({ event });
-    }
-  };
-
-  handleFocus = (event: SyntheticFocusEvent<HTMLDivElement>) => {
-    const { disabled, onFocus } = this.props;
-    if (!disabled && onFocus) {
-      onFocus({ event });
-    }
-  };
-
-  handleMouseEnter = (event: SyntheticMouseEvent<HTMLDivElement>) => {
-    const { disabled, onMouseEnter } = this.props;
-    if (!disabled && onMouseEnter) {
-      onMouseEnter({ event });
+export default function Touchable({
+  accessibilityControls,
+  accessibilityExpanded,
+  accessibilityHaspopup,
+  accessibilityLabel,
+  children,
+  disabled = false,
+  fullWidth = true,
+  fullHeight,
+  mouseCursor = 'pointer',
+  onBlur,
+  onFocus,
+  onMouseEnter,
+  onMouseLeave,
+  onTouch,
+  pressBackgroundEnabled = false,
+  pressScaleEnabled = false,
+  rounding = 0,
+}: Props) {
+  const [isFocused, setFocused] = React.useState<boolean>(false);
+  const [isPressed, setPressed] = React.useState<boolean>(false);
+  const [coordinate, setCoordinate] = React.useState<Coordinate>({
+    x: 0,
+    y: 0,
+  });
+  const ignoreNextFocusEventRef = React.useRef<boolean>(false);
```

Comment:
IIUC, this is to handle undesired focus event fired immediately after touch ends. See https://phabricator.pinadmin.com/D213915 for details.